src/emitter.hpp

```diff
@@ -87,37 +88,31 @@ class Emitter {
   /**
    * @brief Construct a new Emitter object
    *
-   * The `db_name` can be any valid path on your host file system (that can be created with the current user).
-   * By default it will create the required files wherever the application is being run from.
-   *
    * @param uri The URI to send events to
    * @param method The request type to use (GET or POST)
    * @param protocol The protocol to use (http or https)
    * @param send_limit The maximum amount of events to send at a time
    * @param byte_limit_post The byte limit when sending a POST request
    * @param byte_limit_get The byte limit when sending a GET request
-   * @param db_name Defines the path and file name of the database
+   * @param event_store Defines the database to use for event queue
    */
   Emitter(const string & uri, Method method, Protocol protocol, int send_limit, 
-    int byte_limit_post, int byte_limit_get, const string & db_name);
+    int byte_limit_post, int byte_limit_get, shared_ptr<EventStore> event_store);
```

Comment:
It would be nice to also provide this constructor, but the problem is that without the Singleton storage instance, we can't initialize the storage for users because it needs to be initialized both here in the Emitter as well as in the ClientSession. If they were each passed the same db_name string, we would initialize two SqliteStorage instances, which means two connections to the SQLite database. That is probably fine but not ideal, I think it's better if users initialize the SqliteStorage themselves and then pass the same instance to Emitter or ClientSession (or use their custom storage).